moped-editor/src/views/projects/projectView/ProjectFiles.js

```diff
@@ -134,278 +334,161 @@ const ProjectFiles = (props) => {
-            <MenuItem value={2}>Plans</MenuItem>
-            <MenuItem value={3}>Estimates</MenuItem>
-            <MenuItem value={4}>Other</MenuItem>
-          </Select>
-          <FormHelperText>Required</FormHelperText>
-        </FormControl>
-      ),
-    },
-    {
-      title: "Description",
-      field: "file_description",
-      render: (record) => <span>{record?.file_description}</span>,
-      editComponent: (props) => (
-        <TextField
-          variant="standard"
-          id="file_description"
-          name="file_description"
-          value={props?.value ?? ""}
-          onChange={(e) => props.onChange(e.target.value)}
-        />
-      ),
-    },
-    {
-      title: "Uploaded by",
-      cellStyle: { fontFamily: typography.fontFamily },
-      render: (record) => (
-        <span>
-          {record?.created_by_user_id
-            ? record?.moped_user?.first_name +
-              " " +
-              record?.moped_user?.last_name
-            : "N/A"}
-        </span>
-      ),
-    },
-    {
-      title: "Date uploaded",
-      cellStyle: { fontFamily: typography.fontFamily },
-      customSort: (a, b) =>
-        new Date(a?.created_at ?? 0) - new Date(b?.created_at ?? 0),
-      render: (record) => (
-        <span>
-          {record?.created_at
-            ? `${formatTimeStampTZType(
-                record.created_at
-              )}, ${makeFullTimeFromTimeStampTZ(record.created_at)}`
-            : "N/A"}
-        </span>
-      ),
+    [rowModesModel]
+  );
+
+  const handleSaveClick = useCallback(
+    (id) => () => {
+      setRowModesModel({ ...rowModesModel, [id]: { mode: GridRowModes.View } });
     },
-    {
-      title: "File size",
-      cellStyle: { fontFamily: typography.fontFamily },
-      customSort: (a, b) => (a?.file_size ?? 0) - (b?.file_size ?? 0),
-      render: (record) => (
-        <span>
-          {record.file_key ? humanReadableFileSize(record?.file_size ?? 0) : ""}
-        </span>
-      ),
+    [rowModesModel]
+  );
+
+  // when a user cancels editing by clicking the X in the actions
+  const handleCancelClick = (id) => () => {
+    setRowModesModel({
+      ...rowModesModel,
+      [id]: { mode: GridRowModes.View, ignoreModifications: true },
+    });
+  };
+
+  // handles row delete
+  const handleDeleteClick = useCallback(
+    (id) => () => {
+      // remove row from rows in state
+      setRows(rows.filter((row) => row.project_file_id !== id));
+
+      deleteProjectFileAttachment({
+        variables: {
+          fileId: id,
+        },
+      })
+        .then(() => refetch())
+        .then(() => setIsDeleteConfirmationOpen(false))
+        .catch((error) => {
+          console.error(error);
+        });
     },
-  ];
+    [rows, deleteProjectFileAttachment, refetch]
+  );
+
+  // saves row update after editing an existing row
+  const processRowUpdate = (updatedRow, originalRow) => {
+    const hasRowChanged = !isEqual(updatedRow, originalRow);
```

Comment:
thanks for preventing the empty activity log rows! 🙏 I did notice an existing bug with file activities again. I moved [the issue for it into the backlog](https://github.com/cityofaustin/atd-data-tech/issues/17597) and made it a snackoo. 🍫

Reply:
this is now patched in this pr. 🚀 🙌